samples/iolib/src/main/cpp/player/SampleBuffer.cpp

```diff
@@ -41,4 +46,73 @@ void SampleBuffer::unloadSampleData() {
     mNumSamples = 0;
 }
 
+class ResampleBlock {
+public:
+    int32_t mSampleRate;
+    float*  mBuffer;
+    int32_t mNumFrames;
+};
+
+void resampleData(const ResampleBlock& input, ResampleBlock* output) {
+    // Calculate output buffer size
+    double temp =
+            ((double)input.mNumFrames * (double)output->mSampleRate) / (double)input.mSampleRate;
+
+    // round up
+    int32_t numOutFrames = (int32_t)(temp + 0.5);
```

Comment:
We iterate thousands of times through the loop. Roundoff error could accumulate.
Maybe add room for a few extra frames just in case.

Reply:
Done.